### PR TITLE
Recursive autotranslation

### PR DESCRIPTION
**Pull request by sven15**
Adding recursive auto translation to master branch. After that we can globally activate the autotranslation button.

Currently patched into cms-test.integreat-app.de and can be tested there.

1. Create two pages
2. set one as child of the other one
3. automatic translate the child
4. expected result: child, parent and tree structure should be translated into all languages

### CODE REVIEW

**Comment by Meash** [comment]
Wait why are we merging from the branch `recursive_autotranslation`?
We usually merge into `master` only from `develop` (and hotfixes).

**Comment by sven15** [comment]
Yes. But develop has some huge upcoming merges and this feature is very important. Thats why I don't want to wait for the tests of all other features/merges.

File: wp-content/plugins/automatic-translation/TranslationManager.php [hunk 1]
```diff
@@ -19,6 +19,9 @@
 	return $output_post;
 });
 
+/*
+ * Handling post and it's translation
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: its*

File: wp-content/plugins/automatic-translation/TranslationManager.php [hunk 2]
```diff
@@ -136,6 +159,53 @@ private function should_create_automatic_translation($current_translation_id) {
...
+		$parent = $wpdb->get_results($query, OBJECT);
+		if( $parent[0]->post_parent )
+			$post_id = $parent[0]->post_parent;
+		
+		$query = "SELECT post_parent FROM $wpdb->posts WHERE ID = '$post_id'";
+		$parent = $wpdb->get_results($query, OBJECT);
+		return $parent[0]->post_parent;
+	}
+	
+	//parameters: post object of child, wpml source language code and target language code
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: there is no benefit from this comment.
Refer to [phpdoc's @param](https://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_tags.param.pkg.html)

File: wp-content/plugins/automatic-translation/TranslationWpmlHelper.php [hunk 3]
```diff
@@ -19,4 +19,37 @@ public function link_wpml($source_post_id, $source_post_type, $translated_post_i
...
+		}
+		
+		$source_parent_id = $result[0]->post_parent;
+		$query = "SELECT trid FROM {$wpdb->prefix}icl_translations WHERE element_id = '$source_parent_id'";
+		$result = $wpdb->get_results($query);
+		$trid = $result[0]->trid;
+		
+		$query = "SELECT element_id FROM {$wpdb->prefix}icl_translations WHERE trid = '$trid' AND language_code = '$target_language_code'";
+		$result = $wpdb->get_results($query);
+		$translation_parent_id = $result[0]->element_id;
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationWpmlHelper.php]: Don't join by foot!!!!(!) See e.g. http://www.w3schools.com/sql/sql_join.asp

Comment (sven15) [inline wp-content/plugins/automatic-translation/TranslationWpmlHelper.php]: Thats what i said?

File: wp-content/plugins/automatic-translation/TranslationManager.php [hunk 4]
```diff
@@ -136,6 +159,53 @@ private function should_create_automatic_translation($current_translation_id) {
...
+	private function translate_parent( $post, $source_language_code, $target_language_code ) {
+		$post_translate = get_post( $parent->post_parent );
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: `$parent` is undefined, did you mean `$post`?

Comment (sven15) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: That's what i meant, but this line is just a leftover of old versions. I'll delete it.

**Pull request closed without merging**